File: skills/search_homes.py
import requests
import json
import re
from urllib.parse import quote

results = []

headers = {
    'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
    'Accept-Language': 'en-US,en;q=0.9',
}

# Try Zillow API endpoint
areas = ['Vancouver WA', 'Ridgefield WA', 'Battle Ground WA', 'Camas WA', 'Washougal WA', 'Brush Prairie WA']

print('=== Attempting Redfin API ===')
for area in areas:
    try:
        # Redfin has a more accessible API
        search_url = f'https://www.redfin.com/stingray/do/location-autocomplete?location={quote(area)}&v=2'
        r = requests.get(search_url, headers=headers, timeout=10)
        print(f'Redfin autocomplete for {area}: status={r.status_code}, len={len(r.text)}')
        if r.status_code == 200:
            text = r.text
            if text.startswith('{}&&'):
                text = text[4:]
            data = json.loads(text)
            print(f'  Parsed: {json.dumps(data, indent=2)[:500]}')
    except Exception as e:
        print(f'  Error: {e}')

print('\n=== Attempting Realtor.com API ===')
for area in areas:
    try:
        city = area.replace(' WA', '').replace(' ', '-')
        url = f'https://www.realtor.com/api/v1/hulk_main_srp?client_id=rdc-x&schema=vesta&query=%7B%22status%22%3A%22for_sale%22%2C%22primary%22%3Atrue%2C%22search_location%22%3A%7B%22location%22%3A%22{quote(area)}%22%7D%2C%22beds_min%22%3A5%2C%22baths_min%22%3A2%2C%22price_max%22%3A600000%2C%22year_built_min%22%3A2017%7D'
        r = requests.get(url, headers=headers, timeout=10)
        print(f'Realtor.com for {area}: status={r.status_code}, len={len(r.text)[:200]}')
    except Exception as e:
        print(f'  Error: {e}')

print('\n=== Trying Google Search for listings ===')
search_queries = [
    'site:zillow.com Vancouver WA 5 bedroom house for sale under 600000 built 2017',
    'site:redfin.com Vancouver WA 5 bedroom 2.5 bath house under 600000',
    'Vancouver WA 5 bedroom house for sale under 600000 built after 2017',
    'Ridgefield WA 5 bedroom house for sale under 600000',
    'Battle Ground WA 5 bedroom house for sale under 600000',
    'Camas WA 5 bedroom house for sale under 600000',
]

for q in search_queries:
    try:
        url = f'https://www.google.com/search?q={quote(q)}&num=10'
        r = requests.get(url, headers=headers, timeout=10)
        print(f'\nGoogle: "{q[:60]}..." status={r.status_code}')
        # Extract snippets
        snippets = re.findall(r'<span[^>]*>([^<]*\$[^<]*)</span>', r.text)
        for s in snippets[:5]:
            print(f'  ${s}')
        # Extract links
        links = re.findall(r'href="(https?://(?:www\.)?(?:zillow|redfin|realtor)\.com/[^"]+)"', r.text)
        for l in links[:5]:
            print(f'  Link: {l}')
    except Exception as e:
        print(f'  Error: {e}')

print('\nDone with initial search.')

File: skills/search_homes.js
const https = require('https');

function fetch(url, headers) {
  return new Promise((resolve, reject) => {
    const urlObj = new URL(url);
    const options = {
      hostname: urlObj.hostname,
      path: urlObj.pathname + urlObj.search,
      method: 'GET',
      headers: headers || {
        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
        'Accept': '*/*'
      },
      rejectUnauthorized: false
    };
    const req = https.request(options, (res) => {
      let data = '';
      if (res.statusCode >= 300 && res.statusCode < 400 && res.headers.location) {
        resolve({redirect: res.headers.location, status: res.statusCode});
        return;
      }
      res.on('data', chunk => data += chunk);
      res.on('end', () => resolve({data, status: res.statusCode}));
    });
    req.on('error', reject);
    req.setTimeout(15000, () => { req.destroy(); reject(new Error('timeout')); });
    req.end();
  });
}

async function main() {
  console.log('=== SEARCHING REDFIN ===');
  
  // First, find Vancouver WA on Redfin by searching
  try {
    const searchUrl = 'https://www.redfin.com/stingray/do/location-autocomplete?location=Vancouver%20WA&v=2';
    const result = await fetch(searchUrl, {
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36',
      'Accept': '*/*',
      'Referer': 'https://www.redfin.com/'
    });
    let raw = result.data;
    if (raw.startsWith('{}&&')) raw = raw.substring(4);
    console.log('Autocomplete response (' + result.status + '): ' + raw.substring(0, 500));
  } catch(e) {
    console.log('Autocomplete error: ' + e.message);
  }
  
  // Try direct Redfin GIS API with broader search using lat/lng bounds for Vancouver WA area
  try {
    const gisUrl = 'https://www.redfin.com/stingray/api/gis?al=1&market=portland&num_homes=100&ord=days-on-redfin-asc&page_number=1&poly=-122.85%2045.55%2C-122.3%2045.55%2C-122.3%2045.85%2C-122.85%2045.85%2C-122.85%2045.55&sf=1,2,3,5,6,7&min_price=0&max_price=600000&num_beds=5&num_baths=2&min_year_built=2017&v=8';
    const result = await fetch(gisUrl, {
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36',
      'Accept': '*/*',
      'Referer': 'https://www.redfin.com/'
    });
    let raw = result.data;
    if (raw.startsWith('{}&&')) raw = raw.substring(4);
    console.log('GIS response status: ' + result.status + ', length: ' + raw.length);
    if (raw.length > 50) {
      try {
        const data = JSON.parse(raw);
        const homes = (data.payload && data.payload.homes) || [];
        console.log('Found ' + homes.length + ' homes in Vancouver WA area');
        homes.forEach(h => {
          const price = (h.price && h.price.value) || h.price || 'N/A';
          const beds = h.beds || 'N/A';
          const baths = h.baths || 'N/A';
          const sqft = (h.sqFt && h.sqFt.value) || h.sqFt || 'N/A';
          const yr = (h.yearBuilt && h.yearBuilt.value) || h.yearBuilt || 'N/A';
          const street = (h.streetLine && h.streetLine.value) || h.streetLine || '';
          const city = h.city || '';
          const zip = h.zip || '';
          const urlPath = h.url || '';
          console.log('  $' + price + ' | ' + beds + 'bd/' + baths + 'ba | ' + sqft + 'sqft | ' + yr + ' | ' + street + ', ' + city + ' | https://www.redfin.com' + urlPath);
        });
      } catch(e) {
        console.log('JSON parse error: ' + e.message);
        console.log('First 300 chars: ' + raw.substring(0, 300));
      }
    } else {
      console.log('Response too short: ' + raw);
    }
  } catch(e) {
    console.log('GIS error: ' + e.message);
  }
}

main().catch(e => console.log('Fatal: ' + e.message));
